eo-maven-plugin/src/main/java/org/eolang/maven/objectionary/ObjectsIndex.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2023 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven.objectionary;
+
+import java.net.URL;
+import java.util.Set;
+import org.cactoos.Scalar;
+import org.cactoos.Text;
+import org.cactoos.iterable.Mapped;
+import org.cactoos.scalar.ScalarOf;
+import org.cactoos.scalar.Sticky;
+import org.cactoos.set.SetOf;
+import org.cactoos.text.Split;
+import org.cactoos.text.TextOf;
+
+/**
+ * The objects index that contains all available eo objects.
+ * <a href="https://github.com/objectionary/home/blob/gh-pages/objectionary.lst">Example</a>
+ * @since 0.29
+ */
+final class ObjectsIndex {
+
+    /**
+     * Cached objects index.
+     */
+    private final Scalar<? extends Set<String>> objects;
+
+    /**
+     * Ctor.
+     */
+    ObjectsIndex() {
+        this(ObjectsIndex.index());
+    }
+
+    /**
+     * Ctor.
+     * @param all All objects index.
+     */
+    ObjectsIndex(final Scalar<? extends Set<String>> all) {
+        this.objects = new Sticky<>(all);
+    }
+
+    /**
+     * Checks whether object index contains the object.
+     * @param name Object name.
+     * @return True if object index contains the object.
+     * @throws Exception If something unexpected happened.
+     */
+    public boolean contains(final String name) throws Exception {
+        return this.objects.value().contains(name);
+    }
+
+    /**
+     * Loads objects index.
+     * @return Objects index as a set of strings.
+     */
+    private static Scalar<Set<String>> index() {
+        return new ScalarOf<>(
```

Comment:
@volodya-lombrozo this can be moved into `ObjectsIndex()` constructor. No need in static method.

Reply:
@mximp I would definitely prefer to leave static method, because it allows to read code faster. How I read the code:
```
//constructor

//constructor

//logic
```

You suggest

```
//constructor
//logic

//constructor

//logic
```

I don't know if you understand me, but my approach _for me_ looks more readable. This question is matter of taste, of corse. Actually we already had such a discussion previously. What do you think if we create an issue and finally decide  which approach to follow in order to avoid such comments in the future?

Reply:
@volodya-lombrozo To me the motivation is quite simple here - we are trying to produce minimum required number of elements. In this particular case to understand what the constructor does I have to drill down into static method which is purely code extraction. Whereas having code declared within constructor I can see the whole logic in one place.

Reply:
@mximp I understand your point. It makes sense, actually. The main problem is that we spend time on discussing and further refactoring. I believe we just have to agree on one thing.